autogpt_platform/backend/backend/data/execution.py

```diff
@@ -250,12 +279,26 @@ async def get_graph_execution_meta(
     return GraphExecutionMeta.from_db(execution) if execution else None
 
 
-async def get_graph_execution(user_id: str, execution_id: str) -> GraphExecution | None:
+async def get_graph_execution(
+    user_id: str,
+    execution_id: str,
+    include_node_executions: bool = False,
+) -> GraphExecution | GraphExecutionWithNodes | None:
     execution = await AgentGraphExecution.prisma().find_first(
         where={"id": execution_id, "isDeleted": False, "userId": user_id},
-        include=GRAPH_EXECUTION_INCLUDE,
+        include=(
+            GRAPH_EXECUTION_INCLUDE_ALL
+            if include_node_executions
+            else GRAPH_EXECUTION_INCLUDE_IO_ONLY
+        ),
+    )
+    return (
+        (
+            GraphExecutionWithNodes if include_node_executions else GraphExecution
+        ).from_db(execution)
+        if execution
+        else None
```

Comment:
Nit: can we just so normal if else here

Reply:
not sure what you mean?

Reply:
```
if not execution:
    return None
elif include_node_execution:
    return GraphExecutionWithNodes.from_db(execution)
else:
    return GraphExecution.from_db(execution)
```

It's just more obvious to read than a nested for a reviewer, lol